### PR TITLE
feat(service-worker): improve ergonomics of the `SwUpdate` APIs

### PR DESCRIPTION
**Pull request by gkalpak**
_NOTE: This is cleaned up version of #43535._

##
See the original PR and individual commits for details.

Closes #43535

### CODE REVIEW

File: aio/content/guide/deprecations.md [hunk 1]
```diff
@@ -32,36 +32,38 @@ v11 - v14
...
-| `@angular/core/testing` | [`async`](#testing)                                                                           | <!--v9--> v12         |
-| `@angular/core/testing` | [`aotSummaries` argument in `TestBed.initTestEnvironment`](#testing)                                                                           | <!--v13--> v14         |
-| `@angular/core/testing` | [`aotSummaries` field of the `TestModuleMetadata` type](#testing)                                                                           | <!--v13--> v14         |
-| `@angular/forms`        | [`FormBuilder.group` legacy options parameter](api/forms/FormBuilder#group)                   | <!--v11--> v14        |
-| `@angular/router`       | [`ActivatedRoute` params and `queryParams` properties](#activatedroute-props)                 | unspecified           |
-| template syntax         | [`/deep/`, `>>>`, and `::ng-deep`](#deep-component-style-selector)                            | <!--v7--> unspecified |
+| Area                      | API or Feature                                                                                | May be removed in     |
```
Comment (IgorMinar) [inline aio/content/guide/deprecations.md:35]: why such big diff? ws?

File: aio/content/guide/service-worker-communications.md [hunk 2]
```diff
@@ -37,7 +37,8 @@ Do this with the `checkForUpdate()` method:
 
 <code-example path="service-worker-getting-started/src/app/check-for-update.service.ts" header="check-for-update.service.ts"></code-example>
 
-This method returns a `Promise` which indicates that the update check completed successfully, though it does not indicate whether an update was discovered as a result of the check. Even if one is found, the service worker must still successfully download the changed files, which can fail. If successful, the `available` event indicates availability of a new version of the application.
+This method returns a `Promise<boolean>` which indicates if an update is available for activation. The check might fail
+which will cause a rejection of the `Promise`.
```
Comment (IgorMinar) [inline aio/content/guide/service-worker-communications.md:41]: one sentence per line please

**Review by IgorMinar — APPROVED** [review]
LGTM after cleanup. Thanks!

Reviewed-for: public-api, docs-packaging-and-releasing, fw-service-worker

File: aio/content/guide/deprecations.md (hunk 1 above)
Comment (gkalpak) [reply aio/content/guide/deprecations.md:35]: Yeah. Only the two `@angular/service-worker` lines (near the bottom) were added. Everything else is whitespace to adjust cell sizes so the table remains aligned.

**Review by AndrewKushnir — APPROVED** [review]
Reviewed-for: public-api

**Review by jessicajaniuk — APPROVED** [review]
LGTM :cookie:

reviewed-for: public-api

**Comment by IgorMinar** [comment]
I overrode the CLA label since the original PR has a CLA.

File: aio/content/guide/deprecations.md (hunk 1 above)
Comment (IgorMinar) [reply aio/content/guide/deprecations.md:35]: aha. thanks.

**Comment by dylhunn** [comment]
This PR currently has merge conflicts with master; mind rebasing?

**Comment by gkalpak** [comment]
The CI failures are not related to this PR and will be fixed by #43714.

**Comment by gkalpak** [comment]
**merge-assistance:** The CLA check was overridden (see https://github.com/angular/angular/pull/43668#issuecomment-932610176), but I had to rebase on master and now the CLA check fails again 😞

**Comment by dylhunn** [comment]
Setting CLA to yes, since the original PR had CLA

**Comment by dylhunn** [comment]
This PR was merged into the repository by commit 0dc45446fe487febaefaf68a928c5a249880f2f3.

**Pull request closed without merging**